recipes/boost/all/conanfile.py

```diff
@@ -373,7 +373,7 @@ def _build_bcp(self):
                 self.run(command)
 
     def _run_bcp(self):
-        with tools.vcvars(self.settings) if self._is_msvc else tools.no_op():
+        with tools.vcvars(self.settings) if self._is_msvc or (self.settings.os == "Windows" and self.settings.compiler == "clang") else tools.no_op():
```

Comment:
add property `self._is_clang_cl`, as it's used in multiple places, and the condition is complicated